app.ts

```diff
@@ -0,0 +1,391 @@
+		if (isFunctionRunning) {
+			return;
+		};
+		isFunctionRunning = true;
+		globals.lastPage = globals.firstPage - 1;
+		globals.firstPage = globals.lastPage - 9;
+		$(".pagination").empty();
+		await pageNumbers(globals.firstPage, globals.lastPage);
+		isFunctionRunning = false;
+	});
+};
+
+// Event listener to prevent some characters to be entered in the input
+$('.search-input').on('keydown', function (e) {
+	if (e.key === 'e'|| e.key ===  'E'|| e.key ===  '.' || e.key ===  '+' || e.key ===  '*'|| e.key ===  '-'){
+		e.preventDefault();
+	}
+	if (e.key === 'Enter') {
+		$('.results-box').trigger('click')
+	};
+});
+
+// In this function wil do the extract the number entered in the search. Then it would take that and calculate the range which should be displayed for the user to click on. 
+$(".search-input").on("input", async function (this: HTMLInputElement, event: any) {
+	if (isFunctionRunning) {
+		return;
+	};
+	isFunctionRunning = true;
+	event.preventDefault();
+	let count = await getRecordCount();
+	let inputNumber = input();
+	let inputNumberInt: any = parseInt(inputNumber);
+	if (inputNumber !== '') {
+		const maxRecords = recordsPerPage();
+		let end = Math.ceil(inputNumberInt / maxRecords) * maxRecords;
+		if (end > (count + 1)) {
+			end = count;
+			globals.currentToID = end;
+		};
+		let start = (end - (maxRecords - 1));
+		globals.currentPage = Math.floor(end / maxRecords);
+		if (inputNumberInt < 1000000 && inputNumberInt > 0) {
+			if (end === 1000000) {
+				end = end - 1;
+			} else null;
+			$('.results-box').remove();
+			$('.search-container').append(`
+				<div class="results-box">
+					<p class="results-select">${start} - ${end}</p>
+				</div>`);
+			$('.results-box').on('click', resultsSelect);
+		} else {
+			$('.results-box').remove();
+			$('.search-container').append(`
+				<div class="results-box">
+					<p class="message">Invalid Input!</p>
+				</div>`);
+		};
+	} else {
+		$('.results-box').remove();
+	};
+
+	isFunctionRunning = false;
+});
+
+// After the range has been returned to the user. The user can click on it and that will show the range of records on the table. 
+async function resultsSelect(event: any) {
+	if (isFunctionRunning) {
+		return;
+	};
+	isFunctionRunning = true;
+	let count = await getRecordCount();
+	let idRange = $('.results-select').text();
+	let rangeArray = null;
+	rangeArray = idRange.split('-');
+	$('.results-box').remove();
+	let startID = parseInt(rangeArray[0]);
+	let endID = parseInt(rangeArray[1]);
+	let maxRecords = recordsPerPage();
+	globals.currentPage = Math.ceil(endID / maxRecords);
+	let pageEnd = Math.ceil(globals.currentPage / 10) * 10;
+	let pageStart = pageEnd - 9;
+	if (endID === count) {
+		startID = ((globals.currentPage - 1) * maxRecords) + 1;
+		endID = count;
+		globals.firstPage = pageStart;
+		globals.lastPage = pageEnd;
+	};
+	await pageNumbers(pageStart, pageEnd);
+	await showRecords(startID, endID);
+	isFunctionRunning = false;
```

Comment:
Why not move the `isFunctionRunning = false;` into the isFunctionRunning check is the showRecords func before the return and take it out here